Sources/Sugar/Helpers/Render+userInfo.swift

```diff
@@ -0,0 +1,91 @@
+import Vapor
+
+extension ViewRenderer {
+
+    /// Renders the template bytes into a view using the supplied `Encodable` object as context and
+    /// makes the `Request` available to the tags by including it in the `userInfo` value.
+    ///
+    /// - Parameters:
+    ///     - path: Path to file contianing raw template bytes.
+    ///     - context: `Encodable` item that will be encoded to `TemplateData` and used as template
+    ///         context.
+    ///     - userInfo: User-defined storage.
+    ///     - req: Request
+    /// - Returns: `Future` containing the rendered `View`.
+    public func render<E>(
+        _ path: String,
+        _ context: E,
+        userInfo: [AnyHashable: Any] = [:],
+        on req: Request
+    ) -> Future<View> where E: Encodable {
+        var userInfo = userInfo
+        userInfo[requestUserInfoKey] = req
+        return render(path, context, userInfo: userInfo)
+    }
+
+    /// Renders the template bytes into a view using the supplied `Encodable` object as context and
+    /// makes the `Request` available to the tags by including it in the `userInfo` value.
+    ///
+    /// - Parameters:
+    ///     - path: Path to file contianing raw template bytes.
+    ///     - userInfo: User-defined storage.
+    ///     - req: Request
+    /// - Returns: `Future` containing the rendered `View`.
+    public func render(
+        _ path: String,
+        userInfo: [AnyHashable: Any] = [:],
+        on req: Request
+    ) -> Future<View> {
+        return render(path, Dictionary<String, String>(), userInfo: userInfo, on: req)
+    }
+}
+
+extension TagContext {
+
+    /// Provides access to the `Request` stored in the `userInfo` dictionary.
+    public var request: Request? {
+        get {
+            return context.userInfo[requestUserInfoKey] as? Request
+        }
+        set {
+            context.userInfo[requestUserInfoKey] = newValue
+        }
+    }
+
+    /// Returns the `Request` stored in the `userInfo` dictionary or throws an error.
+    ///
+    /// - Returns: the stored `Request`.
+    /// - Throws: TagContextError.requestNotPassedIntoRender.
+    public func requireRequest() throws -> Request {
+        guard let request = request else {
+            throw TagContextError.requestNotPassedIntoRender
+        }
+        return request
+    }
+}
+
+private let requestUserInfoKey = "_submissions:request"
```

Comment:
Should we name this more generic (and not include "submissions" here)?

Reply:
agreed, that was a copy/paste error. If you agree with the change please accept this suggestion:

```suggestion
private let requestUserInfoKey = "_sugar:request"
```

Reply:
👍